File: src/Runtime/Value/ValueNumber.cpp
#include "ValueNumber.h"

using namespace peak::interpreter;

ValueNumber::ValueNumber(double value)
	: _value(value) {
}

double ValueNumber::GetValue() const {
	return _value;
}
std::string ValueNumber::ToString() const {
	char ch[24]{0};
	snprintf(ch, sizeof(ch), "%0.15lg", _value);
	// sprintf(ch, "%0.15lg", _value);
	return ch;
}

std::shared_ptr<Value> ValueNumber::Clone() const {
	return std::shared_ptr<Value>(new ValueNumber(_value));
}

File: src/Runtime/Value/ValueNumber.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {
	public class ValueNumber : TypeValue<ValueNumber> {
		private double _value = 0;
		public double value { get => _value; }
		public ValueNumber(double value) {
			_value = value;
		}
		public override string ToString() {
			return _value.ToString();
		}
		public override Value Clone() {
			return new ValueNumber(_value);
		}
	}
} // namespace peak.interpreter